src/components/Emoticon/Emoticon.js

```diff
@@ -0,0 +1,105 @@
+import * as React from 'react'
+import PropTypes from 'prop-types'
+import getValidProps from '@helpscout/react-utils/dist/getValidProps'
+import { classNames } from '../../utilities/classNames'
+import { noop } from '../../utilities/other'
+import { EmoticonUI, IconUI } from './Emoticon.css'
+import { getName, getIcon } from './Emoticon.utils'
+
+export class Emoticon extends React.PureComponent {
+  static defaultProps = {
+    center: false,
+    clickable: true,
+    inline: false,
+    isActive: true,
+    isDisabled: false,
+    innerRef: noop,
+    name: 'happy',
+    role: 'presentation',
+    size: 'md',
+    title: '',
+  }
+
+  static className = 'c-Emoticon'
+
+  getClassNames() {
+    const {
+      className,
+      clickable,
+      center,
+      isActive,
+      isDisabled,
+      inline,
+      name,
+      size,
+    } = this.props
+
+    return classNames(
+      Emoticon.className,
+      !clickable && 'is-noInteract',
+      isActive ? 'is-active' : 'is-inactive',
+      center && 'is-center',
+      inline && 'is-inline',
+      isDisabled && 'is-disabled',
+      name && `is-${getName(name)}`,
+      size && `is-${size}`,
+      className
+    )
+  }
+
+  render() {
+    const {
+      className,
+      center,
+      clickable,
+      innerRef,
+      isActive,
+      isDisabled,
+      inline,
+      name,
+      title,
+      size,
+      ...rest
+    } = this.props
+    const iconName = getName(name)
+    const icon = getIcon(iconName, size)
+    const src = { __html: icon }
+
+    return (
+      <EmoticonUI
+        {...getValidProps(rest)}
+        size={size}
+        className={this.getClassNames()}
+        innerRef={innerRef}
+      >
+        <IconUI
+          className="c-Emoticon__icon"
+          dangerouslySetInnerHTML={src}
+          title={title}
+        />
+      </EmoticonUI>
+    )
+  }
+}
+
+Emoticon.propTypes = {
```

Comment:
any particular reason why the propTypes declaration is not inside the component?